adafruit_ble/characteristics/__init__.py

```diff
@@ -127,9 +125,9 @@ def __bind_locally(self, service, initial_value):
             initial_value = bytes(self.max_length)
         max_length = self.max_length
         if max_length is None and initial_value is None:
-            max_length = 20
-            initial_value = bytes(max_length)
-        if max_length is None:
+            max_length = 0
+            initial_value = b""
```

Comment:
`max_length` of 0 is not very useful. I think this should throw an error if both `max_length` and `initial_value` are unspecified.

I was worried that `UARTService` might break also because it was assuming a default of 20, but I see that `ComplexCharacteristic`still has a default of 20. Should `ComplexCharacteristic` set default `length=None` but also a similar check and not allow `None` for both `length` and `initial_value`? I think so. Then `StreamIn` and `StreamOut` would explicitly pass in `20`, or have add an optional `max_length=20` parameter, or maybe pass in all the length-related parameters.

And other services that assume a default of 20 should be checked and modified if necessary.

Reply:
So, I started tweaking this but I think it's a bigger rabbit hole than it first appears. I'd rather wait to tackle this later with BLE MIDI.

It is a rabbit hole because the 20 bytes is knowledge that this layer shouldn't actually have because it varies with the MTU. Hardcoding it here limits the throughput of stream characteristics. I agree there should be a check that initial value or max length is set for fixed characteristics but it's best put in _bleio where the stack has knowledge of its limits.

I confirmed that the Color Picker demo still works due to the 20 default in ComplexCharacteristic.